<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>831</width>
    <height>715</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Parser de Emotibit</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::SystemLogOut"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>10</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Cargar Archivo de captura Emotibit</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelArchivo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>331</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Archivo en bruto EmotiBit:   __________________________________</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>60</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Parser</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>621</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>convierte los paquetes de datos en estructuras comprensibles por el sistema de visualización</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>601</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>La clase DataParser se encarga de procesar y estructurar los datos crudos recibidos vía UDP,</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>451</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>26</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Parser De EmotiBit</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>811</width>
      <height>511</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Archivos</string>
     </attribute>
     <widget class="QTextBrowser" name="textBrowser">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>801</width>
        <height>471</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Información</string>
     </attribute>
     <widget class="QTextEdit" name="textEdit">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>801</width>
        <height>441</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:700; color:#aa0000;&quot;&gt;Aspectos destacados de los archivos generados&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Un archivo por canal&lt;/span&gt;&lt;br /&gt;Cada canal registrado en el archivo de entrada (como &lt;span style=&quot; font-family:'Courier New';&quot;&gt;EA&lt;/span&gt;, &lt;span style=&quot; font-family:'Courier New';&quot;&gt;AX&lt;/span&gt;, &lt;span style=&quot; font-family:'Courier New';&quot;&gt;GY&lt;/span&gt;, etc.) da lugar a un archivo CSV independiente. Esto facilita el análisis individualizado por señal.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#0000ff;&quot;&gt;Nombres descriptivos y ordenados&lt;/span&gt;&lt;br /&gt;Los archivos de salida se nombran con el patrón:&lt;br /&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;nombreBase_canal.CSV&lt;/span&gt; (por ejemplo: &lt;span style=&quot; font-family:'Courier New';&quot;&gt;2_8_2025_EA.CSV&lt;/span&gt;),&lt;br /&gt;y se guardan en el mismo directorio que el archivo original. Esto ayuda a mantener una estructura organizada y coherente.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Formato de datos estandarizado&lt;/span&gt;&lt;br /&gt;Cada archivo tiene dos columnas:&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;
&lt;li style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;time&lt;/span&gt;: tiempo relativo (en segundos) respecto a la primera marca temporal detectada.&lt;/li&gt;
&lt;li style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New';&quot;&gt;&amp;lt;canal&amp;gt;&lt;/span&gt;: valor numérico de la muestra.&lt;br /&gt;Esto permite cargar fácilmente los datos en herramientas externas como Python (Pandas), MATLAB, Excel o R.&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Orden cronológico garantizado&lt;/span&gt;&lt;br /&gt;Las muestras dentro de cada archivo están &lt;span style=&quot; font-weight:700;&quot;&gt;ordenadas por tiempo creciente&lt;/span&gt;, incluso si los datos originales llegaban en paquetes fuera de orden.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Datos filtrados y verificados&lt;/span&gt;&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;
&lt;li style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Se omiten líneas mal formateadas o incompletas.&lt;/li&gt;
&lt;li style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Solo se procesan los canales definidos en la clase &lt;span style=&quot; font-family:'Courier New';&quot;&gt;ChannelFrequencies&lt;/span&gt;, lo que evita archivos vacíos o irrelevantes.&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Tiempos corregidos por frecuencia&lt;/span&gt;&lt;br /&gt;Los tiempos de cada muestra se reconstruyen según la frecuencia real de muestreo de cada canal, calculando el instante de cada dato dentro del paquete. Al inicio de cada registro &lt;span style=&quot; font-weight:700;&quot;&gt;pueden haber valores negativos de tiempo&lt;/span&gt; de pocos milisegundos. Esto se debe interpretar como datos recibidos despues del inicio de la captura, pero generados justo un instante antes, se trata de paquetes de datos en transito en el momento de inicio de la captura. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>831</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
